src/sas/qtgui/Utilities/TabbedModelEditor.py

```diff
@@ -432,24 +425,47 @@ def updateFromEditor(self):
         if not w.is_python:
             base, _ = os.path.splitext(filename)
             filename = base + '.c'
-
         # make sure we have the file handle ready
-        assert(filename != "")
+        assert filename != ""
```

Comment:
The python "optimizer" suppresses assert statements so technically we shouldn't rely on it for flow control. In practice we never run with the optimizer so not a problem. In this case we would fail on self.writeFile with a missing filename so even less of an issue.

Reply:
This is a valid complaint. We should not be using `assert` for type/value checks but instead perform explicit checks and notify the user with relevant message.
Yes - we already have quite a few asserts sprinkled all over the (non-test) code but I think we should be more "vocal" wrt. error checks.